File: src/main/java/io/alexjoest/stackupup/mixin/early/RenderItemMixin.java
package io.alexjoest.stackupup.mixin.early;

import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
import io.alexjoest.stackupup.client.StackRenderHooks;
import net.minecraft.client.gui.FontRenderer;
import net.minecraft.client.renderer.RenderItem;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;

@Mixin(RenderItem.class)
public abstract class RenderItemMixin {
    @WrapOperation(
        method = "renderItemOverlayIntoGUI(Lnet/minecraft/client/gui/FontRenderer;Lnet/minecraft/item/ItemStack;IILjava/lang/String;)V",
        at = @At(
            value = "INVOKE",
            target = "Lnet/minecraft/client/gui/FontRenderer;drawStringWithShadow(Ljava/lang/String;FFI)I"
        )
    )
    private int stackupup$drawLargeCount(
        FontRenderer fontRenderer,
        String text,
        float x,
        float y,
        int color,
        Operation<Integer> original
    ) {
        return StackRenderHooks.drawItemCountWithShadow(fontRenderer, text, x, y, color);
    }
}

File: src/main/kotlin/pl/asie/stackup/mixin/early/RenderItemMixin.kt
package pl.asie.stackup.mixin.early

import net.minecraft.client.gui.FontRenderer
import net.minecraft.client.renderer.RenderItem
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Redirect
import pl.asie.stackup.client.StackUpClientHelpers

@Mixin(RenderItem::class)
abstract class RenderItemMixin {
    @Redirect(
        method = ["renderItemOverlayIntoGUI", "func_180453_a"],
        at = At(
            value = "INVOKE",
            target = "Lnet/minecraft/client/gui/FontRenderer;drawStringWithShadow(Ljava/lang/String;FFI)I"
        )
    )
    private fun drawLargeCount(fr: FontRenderer, text: String, x: Float, y: Float, color: Int): Int {
        return StackUpClientHelpers.drawItemCountWithShadow(fr, text, x, y, color)
    }
}
